### PR TITLE
Update tests and CI Workflows

### PR DESCRIPTION
**Pull request by SirSevenG**
* Updated CC tests (switched to decimal only, updated some tests logic)
* Reanimated OSX CI flow
* Running tests on Windows
* Separate CI workflows for each OS
* triggers only on PR

### CODE REVIEW

File: .github/workflows/komodo_linux_ci.yml [hunk 1]
```diff
@@ -247,32 +224,41 @@ jobs:
...
-        sudo update-alternatives --set x86_64-w64-mingw32-gcc /usr/bin/x86_64-w64-mingw32-gcc-posix
-        sudo update-alternatives --set x86_64-w64-mingw32-g++ /usr/bin/x86_64-w64-mingw32-g++-posix
-        CONFIGURE_FLAGS='CPPFLAGS=-DTESTMODE' ./zcutil/build-win.sh -j$(nproc)
-        zip --junk-paths komodod_win src/komodod.exe
-    - name: Upload komodod.exe as artifact
-      uses: actions/upload-artifact@v1
-      with:
-        name: komodod_win
-        path: ./komodod_win.zip
+#  linux-test-dexp2p:
```
Comment (tonymorony) [inline .github/workflows/komodo_linux_ci.yml]: @SirSevenG  is dexp2p broken?

File: zcutil/build-mac-dtest.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
```
Comment (tonymorony) [inline zcutil/build-mac-dtest.sh]: @SirSevenG Why do we need these files - just to export compilation flag?

File: .github/workflows/komodo_linux_ci.yml (hunk 1 above)
Comment (SirSevenG) [reply .github/workflows/komodo_linux_ci.yml]: Was dexp2p merged to dev already?

File: zcutil/build-mac-dtest.sh (hunk 2 above)
Comment (SirSevenG) [reply zcutil/build-mac-dtest.sh]: Don't want to change build scripts, and only linux script `build.sh`adds to ./configure compilation flags from ENV.

File: .github/workflows/komodo_linux_ci.yml (hunk 1 above)
Comment (SirSevenG) [reply .github/workflows/komodo_linux_ci.yml]: @tonymorony Just re-checked - dexp2p is not in dev branch yet.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -41,7 +41,9 @@ If you don't see your idea listed, and you think it can contribute to Komodo, do
 * **If your contribution is minor,** such as a fixing a typo, open a pull request.
 * **If your contribution is major,** such as a new feature or bugfix, start by opening an issue first. That way, other contributors can weigh in on the discussion before you do any work.
 
-## Style guide
+## Code style and standard rules guide
```
Comment (tonymorony) [inline CONTRIBUTING.md]: In general, these changes shouldn't be in this PR 
However those looks like a fixes

**Comment by tonymorony** [comment]
@SirSevenG CI failing now, I wonder this https://github.com/KomodoPlatform/komodo/pull/312/commits/634246a1ce7e0821f619bb65e5eeb255ad353a96 commit needs to be added to your dev branch (= to this PR)

**Comment by SirSevenG** [comment]
Closing for now, will open a new PR after cleaning up local dev branch.

**Pull request closed without merging**